plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/RowPredicate.java

```diff
@@ -29,13 +29,14 @@ default RowPredicate and(RowPredicate other)
         return (page, position) -> test(page, position) && other.test(page, position);
     }
 
-    default Page filterPage(Page page)
+    default Page filterPage(Page page, int[] columns)
     {
         int positionCount = page.getPositionCount();
         int[] retained = new int[positionCount];
         int retainedCount = 0;
         for (int position = 0; position < positionCount; position++) {
-            if (test(page, position)) {
+            Page selectedPage = columns.length == 0 ? page : page.getColumns(columns);
+            if (test(selectedPage, position)) {
```

Comment:
If we extract a subset of columns as input to `test`, then does it break assumptions about block index in the implementations ? E.g. in `io.trino.plugin.iceberg.delete.PositionDeleteFilter#createPredicate` we have `page.getBlock(filePosChannel)` and that can select different column if we pass in a subset of blocks of the page.

Reply:
I'm not sure if I fully understand your concern. `List<IcebergColumnHandle> columns` argument in `PositionDeleteFilter#createPredicate` is also adjusted in this PR. https://github.com/trinodb/trino/pull/24512/files#diff-705548e3717786bbc076aba181f1eca081ec69200050c18100bfa959e05095dfR357